prow/oss/terraform/modules/alerts/main.tf

```diff
@@ -12,6 +12,44 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
+resource "google_monitoring_alert_policy" "ci-absent" {
+  for_each     = var.prow_components
+  project      = var.project
+  display_name = "ci-absent-${each.key}"
+  combiner     = "OR" # required
+
+  conditions {
+    display_name = "ci-absent-${each.key}"
+
+    condition_monitoring_query_language {
+      duration = "600s"
+      query    = <<-EOT
+      fetch k8s_container
+      | metric 'kubernetes.io/container/restart_count'
+      | filter
+          (resource.container_name == '${each.key}' && resource.namespace_name == '${each.value.namespace}')
+      | align rate(1m)
+      | every 1m
+      | group_by [], [row_count: row_count()]
+      | condition val() > 10 '1'
+      EOT
+      trigger {
+        count = 1
+      }
+    }
+  }
+
+  documentation {
+    content   = "The service `${each.key}` has been restartingmore than 10 times for 10 minutes."
```

Comment:
IIUC this query is actually checking if there are at least 10 restarts **per minute** for 10 minutes. Is that correct? If so I suspect that threshold is too high since crash loop backoff will increase the retry delay.